Tests/Core/OperationTests.swift

```diff
@@ -379,4 +391,44 @@ class DelayOperationTests: OperationTests {
     }
 }
 
+class CancellationOperationTests: OperationTests {
+
+    func test__operation_with_dependency_cancelled_before_adding_still_executes() {
+
+        let delay = DelayOperation(interval: 2)
+        delay.log.severity = .Verbose
+
+        let operation = TestOperation()
+        operation.log.severity = .Verbose
+
+        operation.addDependency(delay)
+        addCompletionBlockToTestOperation(operation, withExpectation: expectationWithDescription("Test: \(__FUNCTION__)"))
+
+        delay.cancel()
+
+        runOperations(delay, operation)
+        waitForExpectationsWithTimeout(5, handler: nil)
+
+        XCTAssertTrue(operation.didExecute)
+    }
+
+
+    func test__operation_with_dependency_cancelled_after_adding_does_not_execute() {
```

Comment:
For this test, the output in the test runner looks like this:

```
Test Case '-[iOS_OperationsTests.CancellationOperationTests test__operation_with_dependency_cancelled_after_adding_does_not_execute]' started.
Delay for 2.0 seconds: Initialized -> Pending
Delay for 2.0 seconds: Pending -> Ready
Delay for 2.0 seconds: Ready -> Executing
Test Operation: Initialized -> Pending
Delay for 2.0 seconds: Did cancel.
Delay for Test Case '-[iOS_OperationsTests.CancellationOperationTests test__operation_with_dependency_cancelled_after_adding_does_not_execute]' passed (0.001 seconds).
2.0 seconds: Will Execute
```

Note that in this case, the delay still enters the executing state. But also that the dependent `TestOperation` never moves from of the `.Pending` state as far as we can tell.